extensions/librdkafka/PublishKafka.cpp

```diff
@@ -184,12 +184,12 @@ class ReadCallback {
     });
   }
 
-  static rd_kafka_headers_unique_ptr make_headers(const core::FlowFile& flow_file, utils::Regex& attribute_name_regex) {
+  static rd_kafka_headers_unique_ptr make_headers(const core::FlowFile& flow_file, std::optional<utils::Regex>& attribute_name_regex) {
     const gsl::owner<rd_kafka_headers_t*> result{ rd_kafka_headers_new(8) };
     if (!result) { throw std::bad_alloc{}; }
 
     for (const auto& kv : flow_file.getAttributes()) {
-      if (utils::regexSearch(kv.first, attribute_name_regex)) {
+      if (attribute_name_regex && utils::regexSearch(kv.first, *attribute_name_regex)) {
```

Comment:
Wherever regexSearch is used over regexMatch, we should check if it really makes sense to use subsequence matching.

Reply:
Good idea, updated in 2c70c9d7f88a3b7568408c2b45c44516b199ae37